bridge/gitlab/iterator.go

```diff
@@ -0,0 +1,253 @@
+package gitlab
+
+import (
+	"time"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type issueIterator struct {
+	page  int
+	index int
+	cache []*gitlab.Issue
+}
+
+type noteIterator struct {
+	page  int
+	index int
+	cache []*gitlab.Note
+}
+
+type labelEventIterator struct {
+	page  int
+	index int
+	cache []*gitlab.LabelEvent
+}
+
+type iterator struct {
+	// gitlab api v4 client
+	gc *gitlab.Client
+
+	// if since is given the iterator will query only the updated
+	// issues after this date
+	since time.Time
+
+	// project id
+	project string
+
+	// number of issues and notes to query at once
+	capacity int
+
+	// sticky error
+	err error
+
+	// issues iterator
+	issue *issueIterator
+
+	// notes iterator
+	note *noteIterator
+
+	// labelEvent iterator
+	labelEvent *labelEventIterator
+}
+
+// NewIterator create a new iterator
+func NewIterator(projectID, token string, since time.Time) *iterator {
+	return &iterator{
+		gc:       buildClient(token),
+		project:  projectID,
+		since:    since,
+		capacity: 20,
```

Comment:
Would it be good to make a `capacity` parameter in `NewIterator` and `gitlab.Init` functions and control that within the bridge command

Reply:
For both all the bridges

Reply:
That make sense. It's a performance parameter, and the good value will be dependent of the context (network ...), so the calling code have better clue about what make sense.